MKOB/kobmain.py

```diff
@@ -0,0 +1,95 @@
+"""
+MIT License
+
+Copyright (c) 2020 PyKOB - MorseKOB in Python
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+"""
+kobmain.py
+
+Handle the flow of Morse code throughout the program.
+"""
+
+from pykob import kob, morse, internet
+import kobconfig as kc
+import kobactions as ka
+import kobstationlist
+import kobkeyboard
+
+mySender = None
+myReader = None
+myInternet = None
+connected = False
+
+kob_latched = True
+keyboard_latched = True
+internet_latched = True
```

Comment:
Is there a state table (or something) that explains what the combination of these three states mean? I see from the code that if two are true the third is set false and an action is performed.

Reply:
> Is there a state table (or something) that explains what the combination of these three states mean? I see from the code that if two are true the third is set false and an action is performed.

Those latch variables indicate whether the respective system components are latched in their circuit closed state. For the key, that means the key's circuit closer is closed. For the keyboard, it means the user has entered a plus sign to close the circuit. For the internet, it means the special mark code of +1 has been detected, meaning the remote station's key is closed.

This all touches on the pesky 'break behavior' issue, which I'm struggling with right now. The edge conditions (which should rarely happen, but always seem to anyway) are driving me nuts! There are so many possible interactions. Once I get this working, I will document it. If for no other reason, so I can understand it later.

Reply:
That really helps - thanks! I'll think about it with you.